config/auth_azure_cli_test.go

```diff
@@ -245,6 +245,30 @@ func TestAzureCliCredentials_DoNotFetchIfTenantIdAlreadySet(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestAzureCliCredentials_DoNotSpecifyTenantIdWithMSI(t *testing.T) {
+	cases := []struct {
+		userName string
+	}{
+		{"systemAssignedIdentity"},
+		{"userAssignedIdentity"},
+	}
+	for _, c := range cases {
+		t.Run(c.userName, func(t *testing.T) {
+			env.CleanupEnvironment(t)
+			os.Setenv("PATH", testdataPath())
+			os.Setenv("FAIL_IF_TENANT_ID_SET", "true")
+			os.Setenv("AZ_USER_NAME", c.userName)
+			os.Setenv("AZ_USER_TYPE", "servicePrincipal")
```

Comment:
Prefer using `t.Setenv` ([doc](https://pkg.go.dev/testing#T.Setenv))

```suggestion
			t.Setenv("PATH", testdataPath())
			t.Setenv("FAIL_IF_TENANT_ID_SET", "true")
			t.Setenv("AZ_USER_NAME", c.userName)
			t.Setenv("AZ_USER_TYPE", "servicePrincipal")
```

Reply:
Awesome, thanks for that tip!